zb-db/src/main/java/io/camunda/zeebe/db/impl/rocksdb/transaction/ForeignKeyChecker.java

```diff
@@ -49,7 +50,8 @@ private void assertForeignKeyExists(
       return;
     }
 
-    keyBuffer.putLong(0, foreignKey.columnFamily().ordinal(), ZeebeDbConstants.ZB_DB_BYTE_ORDER);
+    final int columnFamilyValue = ((EnumValue) foreignKey.columnFamily()).getValue();
```

Comment:
:thinking_face: do we need to cast?

Reply:
I believe it's necessary. The alternative is worse, see aa61c8b82d9c8e95bcc7886bc33c7e6e1dee0a0b for details